src/frontend/js/hooks/useCourseProductUnion/index.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { IntlProvider } from 'react-intl';
+import fetchMock from 'fetch-mock';
+import { PropsWithChildren } from 'react';
+import { CourseListItem, CourseProductRelation } from 'types/Joanie';
+import { History, HistoryContext } from 'hooks/useHistory';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+import { SessionProvider } from 'contexts/SessionContext';
+
+import { getRoutes } from 'api/joanie';
+import { mockPaginatedResponse } from 'utils/test/mockPaginatedResponse';
+import { CourseListItemFactory, CourseProductRelationFactory } from 'utils/test/factories/joanie';
+import { useCourseProductUnion } from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  // FIXME: mockRichieContextFactory import is undefined !!!
```

Comment:
This issue appears with factory refactoring ?

Reply:
I don't know it's only in this specific test, it work well in others.

Reply:
Aucune idée de pourquoi ca casse ici spécifiquement 
```sh
 FAIL  js/hooks/useCourseProductUnion/index.spec.tsx
  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'RichieContextFactory')

      40 | jest.mock('utils/context', () => ({
      41 |   __esModule: true,
    > 42 |   default: mockRichieContextFactory({
         |            ^
      43 |     authentication: {
      44 |       endpoint: 'https://demo.endpoint',
      45 |       backend: 'fonzie',

      at mockRichieContextFactory (js/hooks/useCourseProductUnion/index.spec.tsx:42:12)
      at Object.require (js/utils/errors/handle.ts:3:1)
      at Object.require (js/utils/react-query/createQueryClient.ts:4:1)
      at Object.require (js/utils/test/createTestQueryClient.ts:4:1)
      at Object.require (js/hooks/useCourseProductUnion/index.spec.tsx:8:1)

```

Reply:
It's an order import issue. You should import `RichieContextFactory` before
`
import { createTestQueryClient } from 'utils/test/createTestQueryClient';
`

Reply:
It work even if i don't understand why :-(

Reply:
From my understanding, I explain like that : 
`createTestQueryClient` imports `createQueryClient` which imports `context`.

You are mocking the context in your test with `mockRichieContextFactory`.
So if you import `mockRichieContextFactory` after `createTestQueryClient`, the context mock is called with a function not yet defined.